pybamm/solvers/base_solver.py

```diff
@@ -640,77 +680,102 @@ def solve(self, model, t_eval=None, external_variables=None, inputs=None):
         # object, restarting the solver at each discontinuity (and recalculating a
         # consistent state afterwards if a dae)
         old_y0 = model.y0
-        solution = None
+        solutions = None
         for start_index, end_index in zip(start_indices, end_indices):
             pybamm.logger.info(
                 "Calling solver for {} < t < {}".format(
                     t_eval_dimensionless[start_index] * model.timescale_eval,
                     t_eval_dimensionless[end_index - 1] * model.timescale_eval,
                 )
             )
-            new_solution = self._integrate(
-                model, t_eval_dimensionless[start_index:end_index], ext_and_inputs
-            )
-            new_solution.solve_time = timer.time()
-            if solution is None:
-                solution = new_solution
+            ninputs = len(ext_and_inputs_list)
+            if ninputs == 1:
+                new_solution = self._integrate(
+                    model,
+                    t_eval_dimensionless[start_index:end_index],
+                    ext_and_inputs_list[0],
+                )
+                new_solutions = [new_solution]
             else:
-                solution.append(new_solution, start_index=0)
+                with mp.Pool(processes=nproc) as p:
```

Comment:
does using `Pool` with `nproc=1` come with any performance losses? If so, should we just have a simple for loop in that case?

Reply:
I actually compared solving an ensemble of models using either a loop over `ìnputs_list` or a call to `map`.

| implementation | average(s) | std(s) |
|:--------------------:|:---------------:|:-------:|
| loop           |      0.174 |  0.007 |
| starmap        |      0.175 |  0.006 |

Stats over 32 models.

however I used the built-in `map` at the time... so not sure about the overhead induced by creating an extra process.